workflow-types/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum Shell {
+    #[serde(rename(deserialize = "fish"))]
+    Fish,
+    #[serde(rename(deserialize = "bash"))]
+    Bash,
+    #[serde(rename(deserialize = "zsh"))]
+    Zsh,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Workflow {
+    pub name: String,
+    pub command: String,
+    pub tags: Vec<String>,
+    pub description: Option<String>,
+    #[serde(default)]
+    pub arguments: Vec<Argument>,
+    pub source: Option<String>,
+    pub author: Option<String>,
+    pub author_url: Option<String>,
+    #[serde(default)]
```

Comment:
On a side note, will there be commands here that can only be run on specific OS's?

Reply:
I don't think yet, but it's definitely something we will need at some point. I don't think it's worth adding from the start though. Since the lack of the field assumes all OS's., it should also be easy to add when we do want to add it.